Disable device check for foreach kernels

wenleix opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #56872 Support auto generation of device check
* **#56871 Disable device check for foreach kernels**
* #56870 Add device_check place holder for functions
* #56869 Fix tensor device in test_kthvalue_overlap

foreach kernels fall back to slow path when tensor are on different devices

Generated by codemod:
```
fastmod '(- func: _foreach.*)' '${1}
  device_check: NoCheck   # foreach kernels fall back to slow path when tensor are on different devices'   aten/src/ATen/native/native_functions.yaml
```

Differential Revision: [D27986560](https://our.internmc.facebook.com/intern/diff/D27986560/)

Commit:
Disable device check for foreach kernels

foreach kernels fall back to slow path when tensor are on different devices

Generated by codemod:
```
fastmod '(- func: _foreach.*)' '${1}
  device_check: NoCheck   # foreach kernels fall back to slow path when tensor are on different devices'   aten/src/ATen/native/native_functions.yaml
```

Differential Revision: [D27986560](https://our.internmc.facebook.com/intern/diff/D27986560/)

[ghstack-poisoned]

Commit:
rebase && fix minor issues on "Disable device check for foreach kernels"


foreach kernels fall back to slow path when tensor are on different devices

Generated by codemod:
```
fastmod '(- func: _foreach.*)' '${1}
  device_check: NoCheck   # foreach kernels fall back to slow path when tensor are on different devices'   aten/src/ATen/native/native_functions.yaml
```

Differential Revision: [D27986560](https://our.internmc.facebook.com/intern/diff/D27986560/)

[ghstack-poisoned]

Commit:
fiix copysign&sparse on "Disable device check for foreach kernels"


foreach kernels fall back to slow path when tensor are on different devices

Generated by codemod:
```
fastmod '(- func: _foreach.*)' '${1}
  device_check: NoCheck   # foreach kernels fall back to slow path when tensor are on different devices'   aten/src/ATen/native/native_functions.yaml
```

Differential Revision: [D27986560](https://our.internmc.facebook.com/intern/diff/D27986560/)

[ghstack-poisoned]

Commit:
fix cudnn&&nonzero on "Disable device check for foreach kernels"


foreach kernels fall back to slow path when tensor are on different devices

Generated by codemod:
```
fastmod '(- func: _foreach.*)' '${1}
  device_check: NoCheck   # foreach kernels fall back to slow path when tensor are on different devices'   aten/src/ATen/native/native_functions.yaml
```

Differential Revision: [D27986560](https://our.internmc.facebook.com/intern/diff/D27986560/)

[ghstack-poisoned]

ezyang reviewed (approved):
Should we add a lint in codegen that asserts that if a function name has foreach in it, its device check is NoCheck? To make sure these are kept consistent.

Commit:
rebase & address comment on "Disable device check for foreach kernels"


foreach kernels fall back to slow path when tensor are on different devices

Generated by codemod:
```
fastmod '(- func: _foreach.*)' '${1}
  device_check: NoCheck   # foreach kernels fall back to slow path when tensor are on different devices'   aten/src/ATen/native/native_functions.yaml
```

Differential Revision: [D27986560](https://our.internmc.facebook.com/intern/diff/D27986560/)

[ghstack-poisoned]

Commit:
rebase && fix special xlogy on "Disable device check for foreach kernels"


foreach kernels fall back to slow path when tensor are on different devices

Generated by codemod:
```
fastmod '(- func: _foreach.*)' '${1}
  device_check: NoCheck   # foreach kernels fall back to slow path when tensor are on different devices'   aten/src/ATen/native/native_functions.yaml
```

Differential Revision: [D27986560](https://our.internmc.facebook.com/intern/diff/D27986560/)

[ghstack-poisoned]
